src/base.js

```diff
@@ -367,18 +361,17 @@ export default class KintoClientBase {
       // from within a batch operation.
       const msg = "This result is generated from within a batch " +
         "operation and should not be consumed.";
-      return Promise.resolve(raw ? { json: msg, headers: { get() {} } } : msg);
+      return raw ? { json: msg, headers: { get() {} } } : msg;
     }
-    const promise = this.fetchServerSettings().then(_ => {
-      return this.http.request(this.remote + request.path, {
-        ...request,
-        body: stringify ? JSON.stringify(request.body) : request.body,
-      });
+    await this.fetchServerSettings();
```

Comment:
This might be outside the scope of this PR, but why do we do this? Is this "priming the pump" for some other behavior that requires it?